### PR TITLE
Activity annotation

### PR DESCRIPTION
**Pull request by alaindargelas**
_What are the reasons/motivation for this change?_

_Explain how this is achieved._

_If applicable, please suggest to reviewers how they can test the change._

### CODE REVIEW

**Comment by alaindargelas** [comment]
@akashlevy , this is Draft

File: passes/sat/sim.cc [hunk 1]
```diff
@@ -1460,7 +1460,7 @@ struct SimWorker : SimShared
 	{
 		log_assert(top == nullptr);
 		fst = new FstData(sim_filename);
-
+        timescale = fst->getTimescaleString();
```
Comment (akashlevy) [inline passes/sat/sim.cc:1463]: Indentation looks wrong. Yosys uses hard tabs with 8 spaces per tab. Make sure that you have tab settings set correctly in VS Code.

Comment (akashlevy) [reply passes/sat/sim.cc:1463]: This applies to this whole PR

File: passes/sat/sim.cc [hunk 2]
```diff
@@ -2397,6 +2397,140 @@ struct VCDWriter : public OutputWriter
 	std::ofstream vcdfile;
 };
 
+struct AnnotateActivity : public OutputWriter
```
Comment (akashlevy) [inline passes/sat/sim.cc:2400]: What is the significance of inheriting from `OutputWriter`?

File: passes/sat/sim.cc [hunk 3]
```diff
@@ -2397,6 +2397,140 @@ struct VCDWriter : public OutputWriter
...
+		std::vector<uint32_t> lastValues;
+		std::vector<uint32_t> toggleCounts;
+		std::vector<uint32_t> highTimes;
+	};
+
+    typedef std::map<int, SignalActivityData> SignalActivityDataMap;
+
+    void write(std::map<int, bool> &use_signal) override
+    {
+	    // Init map
```
Comment (akashlevy) [inline passes/sat/sim.cc:2414]: Needs more comments. Hard for me to understand what is going on.

File: passes/sat/sim.cc (hunk 2 above)
Comment (alaindargelas) [reply passes/sat/sim.cc:2400]: It executes in the natural flow of the sim command, at the end, when writing all the output files. 
It minimizes the intrusive changes to the sim code.

Comment (akashlevy) [reply passes/sat/sim.cc:2400]: Got it! Good idea

**Pull request closed without merging**